packages/mds-geography-author/api.ts

```diff
@@ -0,0 +1,320 @@
+      }
+    }
+  )
+
+  app.post(
+    pathsFor('/geographies/'),
+    checkAccess(scopes => scopes.includes('geographies:write')),
+    async (req, res) => {
+      const geography = req.body
+
+      try {
+        const details = geographyValidationDetails(geography)
+        if (details) {
+          throw new ValidationError(JSON.stringify(details))
+        }
+
+        const recorded_geography = await db.writeGeography(geography)
+        return res.status(201).send(recorded_geography)
+      } catch (err) {
+        await log.error('POST /geographies failed', err.stack)
+        if (err.code === '23505') {
+          return res
+            .status(409)
+            .send({ result: `geography ${geography.geography_id} already exists! Did you mean to PUT?` })
+        }
+        if (err instanceof ValidationError) {
+          return res.status(400).send({ error: err })
+        }
+        /* istanbul ignore next */
+        /* istanbul ignore next */
+        return res.status(500).send({ error: err })
+      }
+    }
+  )
+
+  app.put(
+    pathsFor('/geographies/:geography_id'),
+    checkAccess(scopes => scopes.includes('geographies:write')),
+    async (req, res) => {
+      const geography = req.body
+      try {
+        const details = geographyValidationDetails(geography)
+        if (details) {
+          throw new ValidationError(JSON.stringify(details))
+        }
+        await db.editGeography(geography)
+        return res.status(201).send(geography)
+      } catch (err) {
+        await log.error('failed to edit geography', err.stack)
+        if (err instanceof NotFoundError) {
+          return res.status(404).send({ result: 'not found' })
+        }
+        if (err instanceof ValidationError) {
+          return res.status(400).send({ error: err })
+        }
+        return res.status(500).send(new ServerError(err))
+      }
+    }
+  )
+
+  app.delete(
+    pathsFor('/geographies/:geography_id'),
+    checkAccess(scopes => scopes.includes('geographies:write')),
+    async (req, res) => {
+      const { geography_id } = req.params
+      try {
+        const isPublished = await db.isGeographyPublished(geography_id)
+        if (isPublished) {
+          throw new AlreadyPublishedError('Cannot delete an already published geography')
+        }
+        try {
+          await db.deleteGeographyMetadata(geography_id)
+        } catch (err) {
+          /* No reason to let this bubble up. It's legit for metadata to not exist, and it
+           * seems wrong to throw an error for deleting metadata when this endpoint is mainly
+           * about deleting geographies.
+           */
+          await log.info(`Unable to delete nonexistent metadata for ${geography_id}`)
+        }
+        await db.deleteGeography(geography_id)
+        return res
+          .status(200)
+          .send({ result: `Successfully deleted geography and geography metadata of id ${geography_id}` })
```

Comment:
Seems odd for the response to say we deleted geography_metadata even if it doesn't exist